tests/rkt_volume_mount_test.go

```diff
@@ -0,0 +1,331 @@
+	}
+
+	return cleanupFuncs
+}
+
+type volumeMountTestCase struct {
+	description string
+	// [image name]:[image patches]
+	images         []imagePatch
+	cmdArgs        string
+	podManifest    *schema.PodManifest
+	expectedExit   int
+	expectedResult string
+}
+
+var (
+	volumeMountTestCasesRecursiveCLI = []volumeMountTestCase{
+		{
+			"CLI: recursive mount read file",
+			[]imagePatch{
+				{
+					"rkt-test-run-read-file.aci",
+					[]string{fmt.Sprintf("--exec=/inspect --read-file --file-name %s", tmpdir2filepathpod)},
+				},
+			},
+			fmt.Sprintf(
+				"--volume=test1,kind=host,source=%s,recursive=true --mount volume=test1,target=%s",
+				tmpdir, tmpdirpathpod,
+			),
+			nil,
+			0,
+			tmpdir2innerfilecontent,
+		},
+	}
+
+	volumeMountTestCasesNonRecursiveCLI = []volumeMountTestCase{
+		{
+			"CLI: read file with non-recursive mount",
+			[]imagePatch{
+				{
+					"rkt-test-run-read-file.aci",
+					[]string{fmt.Sprintf("--exec=/inspect --read-file --file-name %s", tmpdir2filepathpod)},
+				},
+			},
+			fmt.Sprintf(
+				"--volume=test,kind=host,source=%s,recursive=false --mount volume=test,target=%s",
+				tmpdir, tmpdirpathpod,
+			),
+			nil,
+			0,
+			tmpdir2outerfilecontent,
+		},
+	}
+
+	volumeMountTestCasesRecursivePodManifest = []volumeMountTestCase{
+		{
+			"Read of nested file for recursive mount",
+			[]imagePatch{
+				{"rkt-test-run-pod-manifest-recursive-mount-stat.aci", []string{}},
+			},
+			"",
+			&schema.PodManifest{
+				Apps: []schema.RuntimeApp{
+					{
+						Name: baseAppName,
+						App: &types.App{
+							Exec:  []string{"/inspect", "--read-file"},
+							User:  "0",
+							Group: "0",
+							Environment: []types.EnvironmentVariable{
+								{"FILE", tmpdir2filepathpod},
+							},
+							MountPoints: []types.MountPoint{
+								{tmpdirmountname, tmpdirpathpod, false},
+							},
+						},
+					},
+				},
+				Volumes: []types.Volume{
+					{tmpdirmountname, "host", tmpdir, nil, &boolTrue, nil, nil, nil},
+				},
+			},
+			0,
+			tmpdir2innerfilecontent,
+		},
+	}
+
+	volumeMountTestCasesNonRecursivePodManifest = []volumeMountTestCase{
+		{
+			"PodManifest: Simple read after write with volume non-recursive mounted in a read-only rootfs.",
+			[]imagePatch{
+				{"rkt-test-run-pod-manifest-read-only-rootfs-vol-rw.aci", []string{}},
+			},
+			"",
+			&schema.PodManifest{
+				Apps: []schema.RuntimeApp{
+					{
+						Name: baseAppName,
+						App: &types.App{
+							Exec:  []string{"/inspect", "--write-file", "--read-file"},
+							User:  "0",
+							Group: "0",
+							Environment: []types.EnvironmentVariable{
+								{"FILE", path.Join(tmpdirpathpod, "file")},
+								{"CONTENT", "host:foo"},
+							},
+							MountPoints: []types.MountPoint{
+								{tmpdirmountname, tmpdirpathpod, false},
+							},
+						},
+						ReadOnlyRootFS: true,
+					},
+				},
+				Volumes: []types.Volume{
+					{tmpdirmountname, "host", tmpdir, nil, nil, nil, nil, nil},
```

Comment:
Isn't this recursive in the coreos flavor and non-recursive in fly? If so it doesn't match the name of the test case